clib.nim

```diff
@@ -4,9 +4,27 @@ import sequtils
 import json
 import logging
 
+var myRegion: MemRegion
+# You just declare a region. Don't have to initialize it. The default 0 values are fine.
+# Regions act like a little mini heap that you use and then throw away when you are done.
+# They don't need to be GC'ed because they just get reset when you are done with them anyway.
+# This is very efficient.
+
 echo "clib initialized"
 
 proc hello_echo*(message:cstring):cstring {.exportc.} =
-  ## Echo a message back
-  result = &"{message} echo"
-  echo &"Nim echo result: {result}"
+  #var myRegion: MemRegion
+  # You could also declare the region here if you wanted.
+  # For this case it doesn't matter, but you can use multiple regions.
+  # It's useful for some use cases to have regions per function or per category, etc...
+  withRegion(myRegion):
```

Comment:
Note: There is another template called `withScratchRegion`, which is supposed to use a default "scratch" region without you having to create your own region. This is the way Araq normally uses it in the tests. @see https://github.com/nim-lang/Nim/blob/devel/tests/gc/thavlak.nim#L422

Your program crashed using the scratchRegion. I am not sure why. I think it may have to do with node-gyp passing incorrect flags to the gcc. I just didn't have time to investigate further.